### DISCUSSION TITLE
Visualea

### DISCUSSION THREAD

**Discussion by thomasarsouze** [openalea/openalea#257] [Development]
From Python2 to Python3 and Qt4 to Qt5/6 and more general developments of Visualea

**Comment by pradal** [comment]
To create a dev environment, install in dev mode, 

- [ ] Create a conda environment with plantgl and pglviewer
```bash
conda create -n visualea -c openalea3 -c conda-forge openalea.plantgl pyqglviewer
conda activate visualea
conda install -c openalea3 -c conda-forge numpy scipy qtconsole pandas matplotlib openalea.sconsx networkx ipykernel ipyparallel
```
- [ ] clone from the openalea org

1. core
2. vpltk
3. grapheditor
4. oalab
5. openalea-components
6. visualea

- [ ] Checkout the visualea branch
- [ ] Install each module using the develop mode
```python setup.py develop```

- [ ] Run `visualea`in a shell or debug it (in VS Code) from `visualeagui.py`

**Reply by pradal** [reply]
@baugetfa Welcome to the VisuAlea discussion with @thomasarsouze

**Comment by baugetfa** [comment]
Fixing uncaught exception that kill the application
see:

- incompatibility with version earlier than PyQt v5.5 [link](https://www.riverbankcomputing.com/static/Docs/PyQt5/incompatibilities.html)
- [some code](https://stackoverflow.com/questions/49065371/why-does-sys-excepthook-behave-differently-when-wrapped) on stackoverflow

What I did commit on visualea:

- commit [029f367](https://github.com/openalea/visualea/pull/14/commits/029f367b02063ca6188734e78fec16902155face) commented `raise e` in decorator that handle error exception. No need to raise something that we caught.
- [6743fa0](https://github.com/openalea/visualea/pull/14/commits/6743fa0e2d775b278ec03a987c9d92f191ecced2) added a `sys.excepthook` at the beginning of visualeagui.py, to catch any exception that passed.

**Comment by pradal** [comment]
# Replace vpltk by [QtPy](https://github.com/spyder-ide/qtpy)

1. First move all the PyQt4 syntax to new syntax

- [x] pyqtSignal to Signal
- [x] pyqtSlot to Slot
- [x] pyqtX* to X*
- [x] QT_VERSION_STR to __version__
- [x] QtGUI.QTabWidget to QtWidgets

When everything is replaced and tested, move to step 2

2. Replace openalea.vpltk.qt by qtpy

**Reply by baugetfa** [reply]
I think everything is ported to qtpy. The routines that were toolkit were moved to the module using them. In the case of `designer.py` that was in `vpltk.qt` it has been moved to `visualea.qt` because:

- it is used only by visualea and oalab
- oalab had existing referencies to visualea 

**Tests**
All tests passed, (did not test sub-modules of openalea-component because it is WIP). However, in oalab the editor in `openalea.oalab.paradigm.controller.ParadigmController` used by default was pyqode that is not compatible with the last pyflakes version. Therefore, an error occurs. To be able to pass tests, I disabled it to only use the oalab editor `openalea.oalab.editor.text_editor.RichTextEditor`. See for example [this issue](https://github.com/PyCQA/pyflakes/issues/733)

**Comment by baugetfa** [comment]
A branch **visualea** has been created again, after the release, in `openalea.visualea` as a dev branch

**Reply by pradal** [reply]
Visualea has been released with deps on grapheditor & the like + stdlib

**Comment by baugetfa** [comment]
**Visualea works on windows 10**
I just tried an install on windows 10, without plantgl (with it it last forever), that went smoothly. The only thing is that visualea is not registered as an exe. But when I run `python visualeagui.py` from `miniconda3\envs\visualea\Lib\site-packages\OpenAlea.Visualea-2.3.3-py3.9.egg\openalea\visualea`, it run.
